sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -2235,6 +2278,58 @@ std::unique_ptr<CurlNetworkConnection> CurlConnectionPool::ExtractOrCreateCurlCo
   return std::make_unique<CurlConnection>(request, options, hostDisplayName, connectionKey);
 }
 
+Azure::Core::Http::_detail::KeepAliveOptions CurlConnection::ParseKeepAliveHeader(
+    std::string const& keepAlive)
+{
+  // Parse the Keep-Alive header to determine if the connection should be kept alive.
+  // The Keep-Alive header is in the format of:
+  // Keep-Alive: timeout=5, max=1000
+  // The timeout is the number of seconds the connection is allowed to be idle before it is closed.
+  // The max is the maximum number of requests that can be made on the connection before it is
+  // closed.
+  // If the header is not present, the connection will be kept alive for the lifetime of the
+  // application.
+  Azure::Core::Http::_detail::KeepAliveOptions keepAliveOptions;
+  std::string const timeoutKey = "timeout=";
+  std::string const maxKey = "max=";
+  auto timeoutPos = keepAlive.find(timeoutKey);
+  auto maxPos = keepAlive.find(maxKey);
+  try
+  {
+
+    if (timeoutPos != std::string::npos)
+    {
+      auto timeoutEnd = keepAlive.find(',', timeoutPos);
+      if (timeoutEnd == std::string::npos)
+      {
+        timeoutEnd = keepAlive.size();
+      }
+
+      keepAliveOptions.ConnectionTimeout = std::chrono::seconds(std::stoi(keepAlive.substr(
```

Comment:
I think I asked this earlier: Does the RFC require that the `timeout` keyword appear before the `max` keyword?

It might be easier to parse the string into separate comma separated elements, and then break each of the elements in to = separated elements. That way you can avoid having to deal with ordering issues at all.

Reply:
i tested all the arrangements and combinations, with added and misisng things

Reply:
Perfect - I'd left the comment before I saw all your test cases.